wz_strategic_plugin/src/wz_strategic_plugin.cpp

```diff
@@ -0,0 +1,508 @@
+
+    // TODO do we need to check for anything before pushing onto the plan?
+    resp.new_plan.maneuvers.push_back(composeLaneFollowingManeuverMessage(
+        current_state.downtrack, traffic_light_down_track, current_state.speed, intersection_speed_.get(),
+        current_state.stamp, light_arrival_time_at_freeflow,
+        lanelet::utils::transform(crossed_lanelets, [](const auto& ll) { return ll.id(); })));
+
+    double intersection_length = intersection_end_downtrack_.get() - traffic_light_down_track;
+
+    ros::Time intersection_exit_time =
+        light_arrival_time_at_freeflow + ros::Duration(intersection_length / intersection_speed_.get());
+
+    resp.new_plan.maneuvers.push_back(composeIntersectionTransitMessage(
+        traffic_light_down_track, intersection_end_downtrack_.get(), intersection_speed_.get(),
+        intersection_speed_.get(), light_arrival_time_at_freeflow, intersection_exit_time, crossed_lanelets.back().id(),
+        nearest_traffic_light->getControlledLanelets().back().id()));
+  }
+  else  // Red or Yellow light
+  {
+    resp.new_plan.maneuvers.push_back(composeStopAndWaitManeuverMessage(
+        current_state.downtrack, traffic_light_down_track, current_state.speed, crossed_lanelets.front().id(),
+        crossed_lanelets.back().id(), current_state.stamp,
+        req.header.stamp + ros::Duration(config_.min_maneuver_planning_period)));
+  }
+}
+
+void WzStrategicPlugin::planWhenWAITING(const cav_srvs::PlanManeuversRequest& req,
+                                        cav_srvs::PlanManeuversResponse& resp, const VehicleState& current_state,
+                                        const std::vector<lanelet::CarmaTrafficLightPtr>& traffic_lights)
+{
+  if (traffic_lights.empty())
+  {
+    throw std::invalid_argument("While in WAITING state, the traffic lights disappeared.");
+  }
+
+  auto nearest_traffic_light = traffic_lights.front();
+
+  double traffic_light_down_track =
+      wm_->routeTrackPos(nearest_traffic_light->stopLine().front().front().basicPoint2d()).downtrack;
+
+  ROS_DEBUG("traffic_light_down_track %f", traffic_light_down_track);
+
+  auto current_light_state_optional = nearest_traffic_light->getState();
+
+  if (!validLightState(current_light_state_optional, ros::Time::now()))
+    return;
+
+  if (current_light_state_optional.get() == lanelet::CarmaTrafficLightState::PERMISSIVE_MOVEMENT_ALLOWED)
+  {
+    transition_table_.signal(TransitEvent::RED_TO_GREEN_LIGHT);  // If the light is green send the light transition
+                                                                 // signal
+    return;
+  }
+
+  // If the light is not green then continue waiting by creating a stop and wait maneuver ontop of the vehicle
+  resp.new_plan.maneuvers.push_back(
+      composeStopAndWaitManeuverMessage(current_state.downtrack - 10.0, traffic_light_down_track, current_state.speed,
```

Comment:
What is the purpose of "-10" here?

Reply:
It's a fixed buffer to add to stopping maneuvers once the vehicle is already stopped to ensure that they can have their
trajectory planned. I made it a constant in code with comment